packages/ibm-products/src/components/Datagrid/types/index.ts

```diff
@@ -180,6 +180,16 @@ interface DataGridTableState
 export interface DataGridTableInstance<T extends object = any>
   extends TableInstance<T> {}
 
+export interface RowAction {
+  id?: string;
+  itemText?: string;
+  icon?: JSXElementConstructor<any>;
```

Comment:
Can we use something like `ComponentType | FunctionComponent` here? This follows what Carbon is using in a lot of their components that include a `renderIcon` prop.